### PR TITLE
#155965580 Add multiple alert methods

### PR DESCRIPTION
**Pull request by meshnesh**
# What does this PR do?

Setting up healthchecks to be able to send out important information concerning the cron jobs through multiple channels

## Description of Task to be completed?

Integrating several alert methods in the system such that the system is able to send a user alerts of all those selected platforms when a cron job is pinged and when it goes down.

## How should this be manually tested?

Open `health checks.io` on your localhost
Navigate to the integrations page.

* For email, click on the add integrations button in the email row, add the email you would like the notifications to be sent to. An email is sent to that email inbox to verify the setup. Click on the button in the email to confirm set up.

* For sms, click on the add integrations button in the sms row. You will be redirected to a page where you can add the phone number you would like to get your notifications on. Add the number and click on the save integration button. The number should be registered with twilio for it to work.

* For slack, click on the add integrations button in the sms row, you are redirected to a page where you 
can select the channel to which the notifications will be sent to from your list of available channels

* For Telegram, click on the add integrations button in the telegram row, you are redirected to a page. Add the callback url, then save it to get notifications

#### Created:
add_sms.html - Added UI for the page where the phone number is added
add_telegram.html - Added UI for the page where the telegram chat id is added
.env - Added the twilio configuration settings locally

#### Edited
hc/api/transports.py - Added the SMS model
hc/front/urls.py - Added the URL that redirects to the sms configuration page
hc/front/views.py - Added the function that loads the page where the number to sms is added
hc/templates/front/channels.html - Added sms to the list of configurable channel options
hc/front/tests/test_add_channel.py

What are the relevant pivotal tracker stories?
[#155965580](https://www.pivotaltracker.com/story/show/155965580)

##### Slack
screen shot 2018-03-28 at 14 02 50

##### SMS
screenshot_20180328-143214

##### Telegram
screen shot 2018-03-29 at 00 46 36

### CODE REVIEW

**Comment by mutoromike** [comment]
Great job integrating the alert methods in the application. Just maybe next time the coverage would increase. Also, try and be including comments in your code to improve readability. Example:
```
+class SMS(HttpTransport):
+    def notify(self, check):
+        account_sid = os.getenv('TWILIO_ACCOUNT_SID')
+        auth_token = os.getenv('TWILIO_AUTH_TOKEN')
+        client = Client(account_sid, auth_token)
+        message = client.messages.create(
+            to=self.channel.value,
+            from_=os.getenv('TWILIO_NUMBER'),
+            body="Healthcheck update\nName: {}\nLast ping: {}\nstatus:{}".format(
+                check.name, check.last_ping.strftime('%x, %X'), check.status)
+        )
```

**Comment by FelixWambiri** [comment]
Great Pull request very detailed.I know time was an issue so I know with time test coverage will improve.

File: hc/api/tests/test_notify.py [hunk 1]
```diff
@@ -215,11 +216,39 @@ def test_victorops(self, mock_post):
...
+        self._setup_data("telegram", "-245525604")
+        mock_post.return_value.status_code = 200
+        date = datetime.now() - timedelta(days=1)
+        self.check.last_ping = datetime.strptime(str(date), "%Y-%m-%d %H:%M:%S.%f")
+        telegram_mocker = Mock()
+        self.channel.notify(self.check, telegram_mocker)
+        assert Notification.objects.count() == 1
+
     ### Test that the web hooks handle connection errors and error 500s
+    @patch("hc.api.transports.requests.request")
```
Comment (frankip) [inline hc/api/tests/test_notify.py:247]: Good work testing error handlers

**Review by frankip — COMMENTED** [review]
good Job on the integrations

**Pull request merged**